### PR TITLE
[feature](invert index) add the inverted index to the file cache

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp [hunk 1]
```diff
@@ -263,7 +264,12 @@ bool DorisCompoundDirectory::FSIndexInput::open(const io::FileSystemSPtr& fs, co
     }
     SharedHandle* h = _CLNEW SharedHandle(path);
 
-    if (!fs->open_file(path, &h->_reader).ok()) {
+    io::FileDescription fd;
+    fd.path = path;
+    io::FileBlockCachePathPolicy cache_policy;
+    auto type = config::enable_file_cache ? config::file_cache_type : "";
+    io::FileReaderOptions reader_options(io::cache_type_from_string(type), cache_policy);
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp:271]: do not create reader_options for each read.

File: be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp [hunk 2]
```diff
@@ -367,7 +373,9 @@ void DorisCompoundDirectory::FSIndexInput::readInternal(uint8_t* b, const int32_
 
     Slice result {b, (size_t)len};
     size_t bytes_read = 0;
-    if (!_handle->_reader->read_at(_pos, result, &bytes_read).ok()) {
+    io::IOContext io_ctx;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp:376]: do not create io_ctx for each read.

**Comment by zzzxl1993** [comment]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Pull request merged**